test/Common.cpp

```diff
@@ -120,6 +122,11 @@ void CommonOptions::validate() const
 		ConfigException,
 		"Invalid test path specified."
 	);
+	assertThrow(
+		!enforceGasTest || evmVersion() == langutil::EVMVersion::latest(),
+		ConfigException,
+		"Gas costs can only be enforced on latest evm version."
```

Comment:
Would it not make sense to support this for all evm versions. E.g. if `EVMVersion` is not defined, it will always run against `latest`. But if we have e.g. `EVMVersion: constantinople` we do the gas measurement exactly for this version, also if we have something like `EVMVersion: >=constantinople`, we could always run against the smallest version defined, e.g. `constantinople` in this case.

Reply:
> > I just checked out your branch and ran the tests locally. It looks like that I have always different gas measurement results.
> 
> It's using `berlin` EVM by default. I'm not sure if that's the default value for `--evm-version` as well. If it's not than that might explain the difference. Can you try to run with `--evm-version berlin`?

maybe the latest version should be default, if not version was not specified.

Reply:
It definitely should. And maybe it is. I just have a suspicion that it might not be (mostly due to https://github.com/ethereum/solidity/pull/10863#discussion_r566815370). If it's not, we should change it but that would be something for a separate PR in my opinion.

Reply:
> But if we have e.g. EVMVersion: constantinople we do the gas measurement exactly for this version, also if we have something like EVMVersion: >=constantinople,

The problem with that is that most tests will probably work only with one specific EVM version. Measurements on other versions will be different and tests will not pass. And we don't really want to put expectations for multiple versions in the isoltest comments (https://gitter.im/ethereum/solidity-dev?at=6012ceb55500a97f82ce5b72).

Reply:
Looks like `berlin` is the default after all: https://github.com/ethereum/solidity/blob/v0.8.1/test/Common.cpp#L182-L193